src/subdomains/generic/gs/gs.service.ts

```diff
@@ -83,6 +92,37 @@ export class GsService {
 
   //*** HELPER METHODS ***//
 
+  private async getUserData(query: SupportDataQuery): Promise<UserData> {
+    if (query.userDataId) {
+      return await this.userDataService.getUserData(+query.userDataId);
+    } else if (query.userAddress) {
+      const user = await this.userService.getUserByAddress(query.userAddress);
+      if (!user) throw new NotFoundException('User not found');
+
+      return user.userData;
+    } else if (query.depositAddress) {
+      const sell = await this.sellService.getSellByDepositAddress(query.depositAddress);
+      if (!sell) throw new NotFoundException('Sell not found');
+
+      return sell.user.userData;
+    } else if (query.iban) {
+      const bankAccount = await this.bankAccountService.getBankAccountByIban(query.iban);
+      if (!bankAccount) throw new NotFoundException('BankAccount not found');
+
+      return bankAccount.user.userData;
+    } else if (query.ref) {
+      const user = await this.userService.getUserByRef(query.ref);
+      if (!user) throw new NotFoundException('User not found');
+
+      return user.userData;
+    } else if (query.bankUsage) {
+      const buy = await this.buyService.getBuyByBankUsage(query.bankUsage);
+      if (!buy) throw new NotFoundException('Buy not found');
+
+      return buy.user.userData;
+    }
+  }
```

Comment:
Maybe remove all the `NotFoundException`, always use `?`s? Check is anyways done at line 76. Something like:
```ts
if (query.userDataId) {
      return await this.userDataService.getUserData(+query.userDataId);
    } else if (query.userAddress) {
      return await this.userService.getUserByAddress(query.userAddress).then((u) => u?.userData);
    } else if (query.depositAddress) {
      return await this.sellService.getSellByDepositAddress(query.depositAddress).then((s) => s?.user.userData);
    } ...
```